src/nip46/utils/validator.ts

```
@@ -0,0 +1,531 @@
+/**
+ * Secure error handler to prevent information disclosure
+ */
+export class SecureErrorHandler {
+  private static securityLogger: Logger | null = null;
+  private static securityLoggingEnabled: boolean = process.env.NODE_ENV !== 'test';
+
+  /**
+   * Initialize security logging with a custom logger
+   */
+  static initializeSecurityLogging(logger?: Logger, enabled: boolean = true): void {
+    this.securityLogger = logger || new Logger({
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)